src/main/java/pl/wojciechkostecki/revel/model/Local.java

```diff
@@ -1,18 +1,20 @@
 package pl.wojciechkostecki.revel.model;
 
+import com.fasterxml.jackson.annotation.JsonManagedReference;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
+import javax.persistence.*;
 import java.time.LocalTime;
 
 @Entity
 @Data
+@EqualsAndHashCode(exclude = "menu")
```

Comment:
Actually I would use this annotation in the child. So, remove local hashcode usage in the menu method. Usually data and rules flows parent -> child. So parent knows what children it has, child doesn't know who his parent is.